src/image-player/image-player.tsx

```diff
@@ -52,10 +58,23 @@ export class ImagePlayer {
     return imagePlayer;
   };
 
-  private _loadImage = (item: ImageItem) => {
+  private _loadImage = (item: ImageItem, attempt = 0) => {
     const img = new Image();
     img.onload = () => {
       item.loaded = true;
+      if (img.width > img.height) {
+        item.portrait = false;
+      } else {
+        item.portrait = true;
+      }
+    };
+    img.onerror = () => {
+      item.errored = true;
```

Comment:
this is correct only after max retry attempts.
also the initialization of `errored` is missing

Reply:
"max retry attempts" may not be reached (if playback reach time of slide before max retry attempts happens). I believe `item.errored = true` should be set until `onload` event happened (image successfully loaded).

Reply:
so at lease you have to set `item.errored = false` once `onload` happened. otherwise, you will move to the single layout even the slide eventually loaded in the next attempts.